include/api.php

```diff
@@ -2888,6 +2888,7 @@ function api_format_items($r, $user_info, $filter_user = false, $type = "json")
 			'favorited' => $item['starred'] ? true : false,
 			'user' =>  $status_user ,
 			'friendica_owner' => $owner_user,
+			'friendica_private' => $item['private'] == 1,
```

Comment:
Why not returning the "private" value directly?

Reply:
because it's a boolean, and in json a boolean is directly encoded and decoded.
and because other booleans are returned as pure booleans (see favorited)

Reply:
Problem is that "private" is not really a boolean but can contain the value "2". I'm unsure if we should treat this differently.

Reply:
It is mainly used for public posts from connector networks. Possibly we don't need this special treatment anymore ... I had to check.

Reply:
When will you be able to check?

Reply:
I found this in core:

```
mod/display.php:250:
$is_public = DBA::exists('item', ['id' => $item_id, 'private' => [0, 2]]);
```

looks like a value of 2 is still public

and

```
mod/item.php:241:
if ($user['hidewall']) {
	$private = 2;
}
```

(that is the only place a quick grep can find private being set to 2)

BUT. in `src/Database/DBStructure.php` we have:

```php
$database["item"] = [
	"comment" => "Structure for all posts",
		"fields" => [
			/* ... */
			"private" => ["type" => "boolean", "not null" => "1", "default" => "0", "comment" => "distribution is restricted"],
```

So it is a boolean. Looks like we have a tri-state boolean, aka "a case of TRUE/FALSE/FILE_NOT_FOUND boolean"

Reply:
Yeah, it is not totally clean. Possibly you are right and we should treat "0" and "2" as public.